### ISSUE TITLE
How to reproduce the metrics in the paper ?

### ISSUE THREAD

**Issue by kingchurch** [Mathux/TMR#8]
Hi Mathux,

Thank you for releasing the TMR training code. I ran the train.py with the default settings against guo's dataset the contrastive_metrics result I got was quite a bit worse than the contrastive_metrics packaged in the pre-trained model.  For example, the following is the content of the "threshold_0.95.yaml" metrics from my newly trained model and the pre-trained model. 

-----------my newly trained model-------------
t2m/R01: 11.43
m2t/R01: 12.27
t2m/R02: 14.03
m2t/R02: 14.32
t2m/R03: 20.0
m2t/R03: 20.51
t2m/R05: 25.84
m2t/R05: 26.05
t2m/R10: 36.11
m2t/R10: 35.45
t2m/MedR: 21.0
m2t/MedR: 23.5
t2m/len: 4384.0
m2t/len: 4384.0

-------------Pre-trained Model--------------
t2m/R01: 13.0
m2t/R01: 12.32
t2m/R02: 16.15
m2t/R02: 14.92
t2m/R03: 21.35
m2t/R03: 21.35
t2m/R05: 28.51
m2t/R05: 27.6
t2m/R10: 39.07
m2t/R10: 37.73
t2m/MedR: 18.0
m2t/MedR: 20.0
t2m/len: 4384.0
m2t/len: 4384.0

The pre-trained model's metrics are clearly better than my newly trained model.  Can you advise how can I reproduce the same level of metrics as the pre-trained model ?  I increased the max_epochs from 1000 to 2000 and it still didn't help. Also I trained the model on a single A6000 GPU with 40GB VRAM. What hardware did you use for the pre-trained model ? Are there other hyper parameters I need change from the default values to achieve the similar results as the pre-trained model ?

Thanks,
Kc

**Comment by Mathux** [comment]
Hello @kingchurch,

Thanks for your issue. As you can see in the pertained log file ``models/tmr_humanml3d_guoh3dfeats/train.out`` or the config file ``models/tmr_humanml3d_guoh3dfeats/config.json`` I trained in only for 500 epochs. I forgot to change "max_epochs" parameter to 500 (instead of 1000) in the general config. I just updated it, this might be the issue.

Can you try to evaluate with the epoch 500 instead? (you can add the argument ``ckpt=latest-epoch=499-v1.ckpt``, it should be saved).

Best

**Comment by kingchurch** [comment]
Hi Mathux,

Thank you for the suggestion. The epoch 500 result looks much better. "threshold_0.95.yaml" as the following.

t2m/R01: 11.72
m2t/R01: 14.14
t2m/R02: 15.88
m2t/R02: 16.1
t2m/R03: 21.24
m2t/R03: 21.42
t2m/R05: 27.49
m2t/R05: 27.21
t2m/R10: 37.98
m2t/R10: 36.98
t2m/MedR: 18.0
m2t/MedR: 21.5
t2m/len: 4384.0
m2t/len: 4384.0

It's still sightly worse than the results from the pre-trained model.  Is this within the expected range of accuracy around 500 epoch ?  Also are there ways like TensorBoard to visualize the training statistics to help us understand when the training result has converged ? How did you pick the max_epochs of 500 ?

Kc

**Comment by LinghaoChan** [comment]
Hi, @kingchurch 

Have you ever tested the `.pth` of earlier checkpoints? When I reproduced the TMR (not based on official codes), the model convergences fast. When I use the TMR, I often choose the TMR of 99 or 199 checkpoints.

**Comment by kingchurch** [comment]
@LinghaoChan Thank you for the suggestion. Yes I found checkpoint 199 provided the best result.
